source/Line.hpp

```diff
@@ -95,12 +97,39 @@ class Line : public io::IO
 	 * @param curv The curvature
 	 */
 	real getNonlinearEI(real curv);
+	
+	/** @brief Finds the depth of the water at some (x, y) point. Either using env->WtrDpth or the 3D seafloor if available
+	 * @param x x coordinate
+	 * @param y y coordinate
+	 * @return A negative number representing the sea floor depth at the given location
+	*/
+	real getWaterDepth(real x, real y) {
+		if (seafloor) {
+			return seafloor->getDepthAt(x, y);
+		}
+		else {
+			return -env->WtrDpth;
+		}
+	}
 
+	/** @brief A single value representing the average water depth
+	 * 
+	*/
+	real avgWaterDepth() {
```

Comment:
This can be clearle an inline function

Reply:
Done. Does explicitly declaring inline have much of an effect these days? [I've been lead to believe it does not](https://stackoverflow.com/questions/1759300/when-should-i-write-the-keyword-inline-for-a-function-method).